sdk/storage/azure-storage-file-share/azure/storage/fileshare/_models.py

```diff
@@ -140,30 +140,30 @@ def _from_generated(cls, generated):
 class ShareSmbSettings(GeneratedShareSmbSettings):
     """ Settings for the SMB protocol.
 
-    :param SmbMultichannel multichannel: Required. Sets the multichannel settings.
+    :keyword SmbMultichannel multichannel: Sets the multichannel settings.
     """
-    def __init__(self, multichannel):
-        self.multichannel = multichannel
+    def __init__(self, **kwargs):
+        self.multichannel = kwargs.get('multichannel')
 
 
 class SmbMultichannel(GeneratedSmbMultichannel):
     """ Settings for Multichannel.
 
-    :param bool enabled: Required. If SMB Multichannel is enabled.
+    :keyword bool enabled: If SMB Multichannel is enabled.
     """
-    def __init__(self, enabled):
-        self.enabled = enabled
+    def __init__(self, **kwargs):
+        self.enabled = kwargs.get('enabled')
 
 
 class ShareProtocolSettings(GeneratedShareProtocolSettings):
     """Protocol Settings class used by the set and get service properties methods in the share service.
 
     Contains protocol properties of the share service such as the SMB setting of the share service.
 
-    :param SmbSettings smb: Required. Sets SMB settings.
+    :keyword SmbSettings smb: Sets SMB settings.
     """
-    def __init__(self, smb):
-        self.smb = smb
+    def __init__(self, **kwargs):
+        self.smb = kwargs.get('smb')
```

Comment:
I think we need to add a check, to make sure smb is set, since it's required if we initialize ShareProtocolSettings, what do you think?

Reply:
So basically add a check that ensures that at least one kwarg is set in `SmbSettings`? correct?

Reply:
yeah, also validate for ShareProtocolSettings and SmbMultichannel